v1.1/certificate-rotation.md

```diff
@@ -0,0 +1,154 @@
+
+### Why CockroachDB generates a combined CA certificate
+
+After completing this process, the nodes have the new CA certificate, and the clients have the new CA certificates as and when they are restarted. But the nodes and client certificates are still signed with the old CA certificate. Thus the nodes and clients are unable to verify each other's identity during inter-node or client-node communication. 
+
+To overcome the issue, we take advantage of the fact that multiple CA certificates can be active at the same time. The node and client certificates can be signed with only the latest CA certificate, but while verifying the identity of another node or a client, they can check with multiple CA certificates uploaded to them. Thus instead of creating only the new certificate while rotating the CA certificates, CockroachDB creates a combined CA certificate that contains the new CA certificate followed by the old CA certificate. As and when node and client certificates are rotated, the combined CA certificate is used to verify old as well as new node and client certificates.
+ 
+### Why rotate CA certificates in advance 
+
+Going back to the CA certificate rotation process, the second step of the process involves rotating the node and client certificates. This involves signing the node certificates with new CA certificates. These new node certificates are uploaded to the nodes as soon as the certs directory on the node is rescanned. But the clients will get their new client certificates signed with the new CA certificates only when they are restarted. The clients won't even have the new CA certificates till they are restarted. Thus node certificates signed by the new CA certificate will not be accepted by clients that do not have the new CA certificate yet. To ensure all nodes and clients have the latest CA certificate, change CA certificates on a completely different schedule, for instance, months before changing the node and client certificates. 
+
+## Steps to rotate CA certificate
+
+### Step 1: Generate new CA certificate and append it to existing CA certificate:
+
+To rotate CA certificates, use the `--overwrite` flag to generate a new CA certificate and key. 
+
+~~~ shell
+$ cockroach cert create-ca --certs-dir=certs --ca-key=ca.new.key --overwrite 
```

Comment:
I think we should change this to two steps:
* rename the old CA key: `mv ca.key ca.old.key`
* generate the new certificate: `cockroach cert create-ca --certs-dir=certs --ca-key=ca.key --overwrite`

This way, the instructions to use `--ca-key=ca.key` to generate node/client certificates are always accurate, even after rotation.